packages/react-router/modules/Route.js

```diff
@@ -19,7 +19,7 @@ class Route extends React.Component {
     const { component, render, match } = props // eslint-disable-line react/prop-types
     let children = props.children  // eslint-disable-line react/prop-types
 
-    if (children && !children.length) children = null
+    if (children && Array.isArray(children) && !children.length) children = null
```

Comment:
Actually, you can just make this `if (Array.isArray(children) && !children.length) children = null`